interpreter/src/main/scala/pl/touk/nussknacker/engine/spel/SpelExpressionSuggester.scala

```diff
@@ -83,20 +88,22 @@ class SpelExpressionSuggester(
             }
             val suggestionsFromFields = filterMapByName(to.fields, p.getName).toList
               .filter { case (fieldName, _) =>
-                // TODO: signal to user that we some values have been filtered
+                // TODO: signal to user that some values have been filtered
                 filterIllegalIdentifierAfterDot(fieldName)
               }
               .map { case (methodName, clazzRef) =>
                 ExpressionSuggestion(methodName, clazzRef, fromClass = false, None, Nil)
               }
             val suggestionsFromClass = clssDefinitions
               .get(to.objType.klass)
-              .map(c =>
-                // TODO: Why is it fromClass = suggestionsFromFields.nonEmpty? instead of just false?
-                filterClassMethods(c, p.getName, staticContext = false, fromClass = suggestionsFromFields.nonEmpty)
-              )
+              .map(c => filterClassMethods(c, p.getName, staticContext = false, fromClass = true))
               .getOrElse(Nil)
-            Future.successful(suggestionsFromFields ++ suggestionsFromClass)
+            val applicableSuggestions = if (nuSpelNodeParentOpt.exists(a => a.spelNode.isInstanceOf[Indexer])) {
```

Comment:
Let's keep this `isInstanceOf` in one place above, we just need to extract a variable let's say `collectSuggestionsFromClass` and we should pass it next to `typedNode`.

Reply:
right, done